CondFormats/HcalObjects/interface/HcalRecoParamWithPulseShapeT.h

```diff
@@ -17,7 +17,55 @@ class HcalRecoParamWithPulseShapeT {
     using PulseShapeConstView = typename PulseShapeCollection::ConstView;
     constexpr ConstView(RecoParamConstView recoView, PulseShapeConstView psView)
         : recoParamView_{recoView}, pulseShapeView_{psView} {};
-    constexpr float pulseShape(int id) const { return 0; }
+
+    constexpr float compute_pulse_shape_value(uint32_t const hashedId,
```

Comment:
In general (after seeing the pixel code) I think prefer the use free functions instead of adding members to views and SoA-like classes.

Reply:
Let me rephrase for clarity: I think I would prefer this to be a free function that takes the `recoParamView()` and `pulseShapeView()` as arguments, rather than as a data member.

Reply:
Done in 11aa37463e89e26d17c31a9e9d76e6f799b8a298

Reply:
After talking to Matti, we came up with better way to implement this. 1dd690878a403817b4a18d302f87bb0db483a05b
We separate the access to pulseShape SoA element (now named `getPulseShape`) from the computation `compute_pulse_shape_value`. 
The compute function is then a free function and is put into `Mahi.dev.cc` alongside the other helper functions.